src/Tribe/Utils/Post_Thumbnail.php

```diff
@@ -0,0 +1,219 @@
+<?php
+/**
+ * Wraps the logic to lazy load a post thumbnail information.
+ *
+ * Example usage:
+ * ```php
+ * $post_thumbnail = new Tribe\Utils\Post_Thumbnail( $post_id );
+ *
+ * // ...some code later...
+ *
+ * // The post thumbnail data is fetched only now.
+ * $large_url = $post_thumbnail->large->url;
+ * ```
+ *
+ * @since   TBD
+ * @package Tribe\Utils
+ */
+
+
+namespace Tribe\Utils;
+
+use Tribe__Utils__Array as Arr;
+
+/**
+ * Class Post_Thumbnail
+ *
+ * @since   TBD
+ * @package Tribe\Utils
+ */
+class Post_Thumbnail implements \ArrayAccess {
+	/**
+	 * An array of the site image sizes, including the `full` one.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected $image_sizes;
+
+	/**
+	 * The post ID this images collection is for.
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	protected $post_id;
+
+	/**
+	 * The post thumbnail data.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected $post_thumbnail_data;
+
+	/**
+	 * Post_Images constructor.
+	 *
+	 * @param int $post_id The post ID.
+	 */
+	public function __construct( $post_id ) {
+		$this->post_id = $post_id;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function __get( $property ) {
+		return $this->offsetGet( $property );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function __set( $property, $value ) {
+		$this->offsetSet( $property, $value );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function __isset( $property ) {
+		return $this->offsetExists( $property );
+	}
+
+	/**
+	 * Fetches and returns the image sizes registered on the site, if any.
+	 *
+	 * @since TBD
+	 *
+	 * @return array An array of the registered image sizes.
+	 */
+	public function get_image_sizes() {
+		if ( null !== $this->image_sizes ) {
+			return $this->image_sizes;
+		}
+
+		$image_sizes = array_merge( [ 'full' ], get_intermediate_image_sizes() );
+
+		/**
+		 * Filters the image sizes the `Tribe\Utils\Post_Thumbnail` class will manage and fetch data for.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $image_sizes All the available image sizes; this includes the default and the intermediate ones.
+		 */
+		$this->image_sizes = apply_filters( 'tribe_post_thumbnail_image_sizes', $image_sizes );
+
+		return $this->image_sizes;
+	}
+
+	/**
+	 * Returns the data about the post thumbnail, if any.
+	 *
+	 * @since TBD
+	 *
+	 * @return array An array of objects containing the post thumbnail data.
+	 */
+	public function get_post_thumbnail_data() {
+		if ( null !== $this->post_thumbnail_data ) {
+			return $this->post_thumbnail_data;
+		}
+
+		$post_id     = $this->post_id;
+		$image_sizes = $this->get_image_sizes();
+
+		$thumbnail_id = get_post_thumbnail_id( $post_id );
+
+		if ( empty( $thumbnail_id ) ) {
+			return [];
+		}
+
+		$thumbnail_data = array_combine(
+			$image_sizes,
+			array_map( static function ( $size ) use ( $thumbnail_id ) {
+				$size_data = wp_get_attachment_image_src( $thumbnail_id, $size );
+
+				if(false === $size_data){
+					return (object)[
```

Comment:
* :no_entry_sign: There must be a space between if and the first parenthesis '('
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space after opening parenthesis is prohibited
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space before closing parenthesis is prohibited
* :no_entry_sign: You must have a space after a closing parenthesis

<sub>hash: a02b1439e9c166c77dffd55a231d0eab</sub>